lib/tasks_private/spec_helper.rake

```diff
@@ -173,20 +165,27 @@ class PopulateTower
     project = create_obj(uri, data)
 
-    # create a job_template
-    uri = '/api/v1/job_templates/'
-    data = {"name" => 'hello_template', "description" => "test job", "job_type" => "run", "project" => project['id'], "playbook" => "hello_world.yml", "credential" => machine_credential['id'], "cloud_credential" => aws_credential['id'], "network_credential" => network_credential['id'], "inventory" => inventory['id'], "organization" => organization['id']}
-    create_obj(uri, data)
-
-    # create a job_template with survey spec
-    uri = '/api/v1/job_templates/'
-    data = {"name" => "hello_template_with_survey", "description" => "test job with survey spec", "job_type" => "run", "project" => project['id'], "playbook" => "hello_world.yml", "credential" => machine_credential['id'], "inventory" => inventory['id'], "survey_enabled" => true, "organization" => organization['id']}
-    template = create_obj(uri, data)
-    # create survey spec
-    uri = "/api/v1/job_templates/#{template['id']}/survey_spec/"
-    data = {"name" => "Simple Survey", "description" => "Description of the simple survey", "spec" => [{"type" => "text", "question_name" => "example question", "question_description" => "What is your favorite color?", "variable" => "favorite_color", "required" => false, "default" => "blue"}]}
-    @conn.post(uri, data).body
-    puts "created #{template['url']} survey_spec"
+    # WARNING! Job template creation may fail if the project hasn't finished cloning.
+    # sleep(10)
+    begin
+      # create a job_template
+      uri = '/api/v1/job_templates/'
+      data = {"name" => 'hello_template', "description" => "test job", "job_type" => "run", "project" => project['id'], "playbook" => "hello_world.yml", "credential" => machine_credential['id'], "cloud_credential" => aws_credential['id'], "network_credential" => network_credential['id'], "inventory" => inventory['id'], "organization" => organization['id']}
+      create_obj(uri, data)
+
+      # create a job_template with survey spec
+      uri = '/api/v1/job_templates/'
+      data = {"name" => "hello_template_with_survey", "description" => "test job with survey spec", "job_type" => "run", "project" => project['id'], "playbook" => "hello_world.yml", "credential" => machine_credential['id'], "inventory" => inventory['id'], "survey_enabled" => true, "organization" => organization['id']}
+      template = create_obj(uri, data)
+      # create survey spec
+      uri = "/api/v1/job_templates/#{template['id']}/survey_spec/"
+      data = {"name" => "Simple Survey", "description" => "Description of the simple survey", "spec" => [{"type" => "text", "question_name" => "example question", "question_description" => "What is your favorite color?", "variable" => "favorite_color", "required" => false, "default" => "blue"}]}
+      @conn.post(uri, data).body
+      puts "created #{template['url']} survey_spec"
+    rescue Faraday::ClientError => error
```

Comment:
First call's failure will now short-circuit the other 2 creations.  If you want to improve this, probably better to
 * put a sleep before this section
 * and/or enclose each individual creation with `rescue` and do a retry for each

Reply:
You’re right and this was my original thought. I somehow got the impression that you’re not a fan of the retry-wait flow for such a one-purpose tool. It was actually not that complex, so I added it instead of this short-circuit. I’ll push it right away.